### PR TITLE
Cover schema management edge cases

### PR DESCRIPTION
**Pull request by morozov**
The newly added index and foreign key introspection tests establish a pattern for testing identifier-related logic.

In the code, we have methods that accept names as objects, as strings interpreted as unquoted names and strings as quoted names. For example: https://github.com/doctrine/dbal/blob/e33aed84e97d4ab28bff365ba95832c56323ad0a/src/Schema/AbstractSchemaManager.php#L775 https://github.com/doctrine/dbal/blob/e33aed84e97d4ab28bff365ba95832c56323ad0a/src/Schema/AbstractSchemaManager.php#L798 https://github.com/doctrine/dbal/blob/e33aed84e97d4ab28bff365ba95832c56323ad0a/src/Schema/AbstractSchemaManager.php#L818

In the test suite, we write one parameterized test that accepts input names as objects and the callbacks wrapping the methods to be tested:

https://github.com/doctrine/dbal/blob/ebbc8d0bfc9a16428956b87db6f989914c61635b/tests/Functional/Schema/SchemaManagerTest.php#L452-L456

They set up the schema using the names and call the method being tested via the callbacks. This way, they exercise the "by name as object" path via the two "by unqualified name" and "by qualified name" entry points. As a result, both kinds of identifiers will be first-class citizens from the test coverage standpoint.

### CODE REVIEW

File: tests/Functional/Schema/SchemaManagerTest.php [hunk 1]
```diff
@@ -446,6 +446,162 @@ public function testDropForeignKey(): void
...
+        callable $introspect,
+    ): void {
+        $this->createTablesWithForeignKey($referencedTableName, $referencingTableName, $indexName, $foreignKeyName);
+
+        self::assertNotEmpty($introspect($this->schemaManager));
+    }
+
+    /**
+     * @return iterable<string, array{
+     *     OptionallyQualifiedName, OptionallyQualifiedName, UnqualifiedName, UnqualifiedName, callable
```
Comment (greg0ire) [inline tests/Functional/Schema/SchemaManagerTest.php]: Should the documentation of `callable` be improved? Like on line 477?

**Review by greg0ire — APPROVED** [review]
